### PR TITLE
Add errors handling when consulting tarification for cares with anato…

### PR DESCRIPTION
**Pull request by hermanju**
The goal of this PR is to handle the errors when consulting tarification for cares with related services and/or anatomy inside the tarif endpoint. The current behavior handles already several errors but I added the one which are not present based on the following documentation: 
[Codes erreur Tarif - Verwerpingscodes Tarief - V1R1 (1).xlsx](https://github.com/user-attachments/files/16158848/Codes.erreur.Tarif.-.Verwerpingscodes.Tarief.-.V1R1.1.xlsx)

Moreover, I fixed a small behavior to handle errors when the xpath is truncated by IO's in the error url ( which is precsiely the case for thoot number missing when consult tarification).

### CODE REVIEW

File: src/main/kotlin/org/taktik/freehealth/middleware/service/impl/TarificationServiceImpl.kt [hunk 1]
```diff
@@ -331,14 +331,14 @@ class TarificationServiceImpl(private val stsService: STSService) : Tarification
                         base = "/${nodeDescr(node.parentNode)}$base"
                         node = node.parentNode
                     }
-                    val elements =
+                    var elements =
```
Comment (aduchate) [inline src/main/kotlin/org/taktik/freehealth/middleware/service/impl/TarificationServiceImpl.kt:334]: Nope... Pas de var

Comment (aduchate) [reply src/main/kotlin/org/taktik/freehealth/middleware/service/impl/TarificationServiceImpl.kt:334]: écris plutôt: val elements = if(elements.isNotEmpty()) ... else ....

**Review by aduchate — CHANGES_REQUESTED** [review]

File: src/main/kotlin/org/taktik/freehealth/middleware/service/impl/TarificationServiceImpl.kt (hunk 1 above)
Comment (hermanju) [reply src/main/kotlin/org/taktik/freehealth/middleware/service/impl/TarificationServiceImpl.kt:334]: Je vais procéder comme suit en utilisant uniquement des val puisque que je ne peux pas utiliser elements pour l'assigner lui-même:

`val initialElements = ConsultTarifErrors.values.filter {
                        it.path == base && it.code == ec && (it.regex == null || url.matches(Regex(".*${it.regex}.*")))
                    }

                    val elements = if (initialElements.isEmpty()) {
                        // IOs sometimes are overeager to provide us with precise xpath. Let's try again while truncating after the item
                        val truncatedBase = base.replace(Regex("(.+/item.+?)/.*"), "$1")
                        ConsultTarifErrors.values.filter {
                            it.path == truncatedBase && it.code == ec && (it.regex == null || url.matches(Regex(".*${it.regex}.*")))
                        }
                    } else {
                        initialElements
                    }`

Comment (hermanju) [reply src/main/kotlin/org/taktik/freehealth/middleware/service/impl/TarificationServiceImpl.kt]: @aduchate J'ai effectué le changement

**Comment by aduchate** [comment]
Hello Julien, could you sign the CLA please ?
https://cla-assistant.io/icure/freehealth-connector?pullRequest=42

**Pull request merged**